src/EnergyPlus/WeatherManager.cc

```diff
@@ -8730,6 +8731,66 @@ Label9999: ;
 
 	}
 
+	void SetWeatherStatistics()
+	{
+		// This function reads the entire weather file for useful statistical
+		// information, e.g.: Annual and monthly averages, etc.
+		int ReadStatus = 0;
+		bool ErrorFound = false;
+		int WYear;
+		int WMonth;
+		int WDay;
+		int WHour;
+		int WMinute;
+		Real64 DryBulb;
+		Real64 DewPoint;
+		Real64 RelHum;
+		Real64 AtmPress;
+		Real64 ETHoriz;
+		Real64 ETDirect;
+		Real64 IRHoriz;
+		Real64 GLBHoriz;
+		Real64 DirectRad;
+		Real64 DiffuseRad;
+		Real64 GLBHorizIllum;
+		Real64 DirectNrmIllum;
+		Real64 DiffuseHorizIllum;
+		Real64 ZenLum;
+		Real64 WindDir;
+		Real64 WindSpeed;
+		Real64 TotalSkyCover;
+		Real64 OpaqueSkyCover;
+		Real64 Visibility;
+		Real64 CeilHeight;
+		Real64 PrecipWater;
+		Real64 AerosolOptDepth;
+		Real64 SnowDepth;
+		Real64 DaysSinceLastSnow;
+		Real64 Albedo;
+		Real64 LiquidPrecip;
+		int PresWeathObs;
+		Array1D_int PresWeathConds( 9 );
+
+		Real64 totalDB = 0.0;
+		int count = 0;
+		std::string WeatherDataLine;
+
+		while (! ReadStatus) {
```

Comment:
There are other places that are currently reading the .stat file or requiring user input which all could get the info directly from a pre-read of the epw file (water mains temperature, adaptive comfort values, FCfactor ground temps, other ground models).  So, this is something good to get in place in some way or another.